user-guide/Reference/Soft-launch_options/BrokerGateway_Migration.md

```diff
@@ -0,0 +1,158 @@
+
+   ```json
+   {
+     "BrokerGatewayConfig": {
+       "CredentialsUrl": "https://LAURENSVG.skyline.local/BrokerGateway/api/natsconnection/getnatsconnectiondetails",
+       "APIKeyPath": "C:\\Program Files\\Skyline Communications\\DataMiner BrokerGateway\\appsettings.runtime.json"
+     }
+   }
+   ```
+
+> [!NOTE]
+> The NATSMigration tool has a **hard-coded 10-minute timeout** during which the *ResetCluster* operation has to be finished. This is the same time frame you have to start the migration (i.e. enter the "install" command in the tool) on all nodes in the cluster.
+
+> [!IMPORTANT]
+> The NATS configuration (*nats-server.config*) of the NATS instance before the migration is not transferrable to the NATS instance after the migration, to it should **never be copied over**.
+
+## Migrating back to the old system
+
+After you have migrated to BrokerGateway, to go back ot the old system, follow the same procedure as when you run the migration, but use the `uninstall` command instead of the `install` command.
+
+The same restrictions apply as for the migration: this must happen on all DataMiner Agents in the cluster at the same time.
+
+## FAQ
+
+### Can I run a cluster with both SLNet-managed NATS and BrokerGateway-managed NATS?
+
+This is not possible. Both NATS installations use the same network ports, so the services cannot run at the same time on a machine. The credentials these installations use are also different and not compatible with each other, so running SLNet-managed NATS on DMA1 and BrokerGateway-managed NATS on DMA2 will also not function.
+
+### Why should I migrate to BrokerGateway? What are the advantages of this?
+
+Careless NATS communications managed by a decoupled and robust BrokerGateway based on a Single Source of Truth that has the complete knowledge of the cluster. A newer and more performant NATS version is also used and is easier than ever to upgrade.
+
+## Troubleshooting
+
+### ERROR: {machineName} was not able to remove itself from its current cluster in order to join the new cluster
+
+When calling *ResetCluster*, BrokerGateway will first try to remove itself from any cluster it is part of, in order to set up a new cluster with all the endpoints specified in *ClusterEndpoints.json*.
+
+The endpoints BrokerGateway is currently clustered with are listed in the ClusterInfo in *C:\Program Files\Skyline Communications\DataMiner BrokerGateway\appsettings.runtime.json*. If one of those IPs cannot be reached, the error message above is generated.
+
+By agreeing to forcibly remove itself from this cluster, the current machine can free itself to cluster with the new setup. However, because it forcibly removes itself, it no longer informs any of these unreachable endpoints of its removal. These may still attempt to contact the current machine, which will no longer be reachable for them.
```

Comment:
Does this forcible removal happen automatically when this error occurs, or does the user have to do something for this?

Reply:
When the error appears the user is asked whether it is ok to forcibly remove itself from the cluster. The user has to enter y/n. On a no, the migration is halted and reverted to the NATSCustodian managed NATS.